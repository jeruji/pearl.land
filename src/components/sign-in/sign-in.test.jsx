import React from "react";
import SignIn from "./sign-in.component";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import { render, cleanup, fireEvent } from "@testing-library/react";

configure({ adapter: new Adapter() });
afterEach(cleanup);
describe("sign-in validator", () => {
  test("When email doesn't match regex, error should be 'Please Check Your Email Address'", () => {
    const email = "notmatch";
    const wrapper = shallow(<SignIn />);
    expect(wrapper.instance().validate("email", email)).toBe(
      "Please Check Your Email Address"
    );
  });
  test('When email does match regex, return should be ""', () => {
    const email = "test@email.com";
    const wrapper = shallow(<SignIn />);
    expect(wrapper.instance().validate("email", email)).toBe("");
  });
});
describe("sign-in submit", () => {
  test("When submit without filling fields should display errors", () => {
    const { getAllByText, getByText } = render(<SignIn />);

    const signInButton = getByText("Sign in");
    fireEvent.click(signInButton);

    const errorEmailSpan = getAllByText("Please Check Your Email Address");
    expect(errorEmailSpan.length).toBe(1);

    const errorPasswordSpan = getAllByText(
      "You Have To Enter At Least 8 Digits"
    );
    expect(errorPasswordSpan.length).toBe(1);
  });
  test("Renders", () => {
    const { container } = render(<SignIn />);
    expect(container).toMatchSnapshot();
  });
  test("When submit after filling in fields should submit okay", () => {
    const { container, getByText, getByLabelText } = render(<SignIn />);

    const emailField = getByLabelText("email");
    expect(emailField).not.toBeNull();
    fireEvent.change(emailField, {
      target: { value: "test@email.com" },
    });

    const passwordField = getByLabelText("password");
    expect(passwordField).not.toBeNull();
    fireEvent.change(passwordField, {
      target: { value: "inipasswordsaya" },
    });

    const signInButton = getByText("Sign in");
    fireEvent.click(signInButton);

    const errorSpan = container.querySelector("[data-testid='formErrors']");
    expect(errorSpan).toBeNull();
  });
});
