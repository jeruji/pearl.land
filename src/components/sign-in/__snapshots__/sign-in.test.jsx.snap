// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign-in submit Renders 1`] = `
<div>
  <div
    class="sign-in"
  >
    <h2>
      I already have an account?
    </h2>
    <span>
      Sign in with your email and password
    </span>
    <form>
      <div
        class="group"
      >
        <input
          aria-label="email"
          class="form-input"
          name="email"
          type="email"
          value=""
        />
        <label
          class=" form-input-label"
        >
          email
        </label>
      </div>
      <div
        class="group"
      >
        <input
          aria-label="password"
          class="form-input"
          name="password"
          type="password"
          value=""
        />
        <label
          class=" form-input-label"
        >
          password
        </label>
      </div>
      <button
        class="custom-button"
        type="submit"
      >
        Sign in
      </button>
    </form>
  </div>
</div>
`;
